src/player.cpp

```diff
@@ -5967,10 +5967,39 @@ void player::suffer()
         }
     }
 
-    if( has_trait( trait_LEAVES ) && g->is_in_sunlight( pos() ) && one_in( 600 ) ) {
+    const auto player_local_temp = g->get_temperature( pos() );
+    double sleeve_factor = armwear_factor();
+    bool has_hat = wearing_something_on( bp_head );
+    bool leafy = has_trait( trait_LEAVES ) || has_trait( trait_LEAVES2 ) || has_trait( trait_LEAVES3 );
+    bool leafier = has_trait( trait_LEAVES2 ) || has_trait( trait_LEAVES3 );
+    bool leafiest = has_trait( trait_LEAVES3 );
+    if( leafy && g->is_in_sunlight( pos() ) ) {
+        int flux = ( player_local_temp - 65 ) / 2;
+        if( !has_hat ) {
+            vitamin_mod( vitamin_id( "counter_sunlight_vitamin" ), ( 100 + flux ), false );
+            vitamin_mod( vitamin_id( "counter_sunlight_kcal" ), ( 100 + flux ), false );
+        }
+        if( leafier ) {
+            int rate = ( ( 100 * sleeve_factor ) + flux ) * 2;
+            vitamin_mod( vitamin_id( "counter_sunlight_vitamin" ), rate, false );
+            vitamin_mod( vitamin_id( "counter_sunlight_kcal" ), rate, false );
+            if( leafiest ) {
+                vitamin_mod( vitamin_id( "counter_sunlight_vitamin" ), rate, false );
+                vitamin_mod( vitamin_id( "counter_sunlight_kcal" ), rate, false );
+            }
+        }
+    }
+
+    if( vitamin_get( vitamin_id( "counter_sunlight_vitamin" ) ) >= 3000 ) {
+        vitamin_mod( vitamin_id( "vitA" ), 1, true );
+        vitamin_mod( vitamin_id( "vitC" ), 1, true );
+        vitamin_set( vitamin_id( "counter_sunlight_vitamin" ), 0 );
+    }
+
+    if( vitamin_get( vitamin_id( "counter_sunlight_kcal" ) ) >= 2000 ) {
         mod_hunger( -1 );
-        // photosynthesis absorbs kcal directly
```

Comment:
why did you remove this comment, if you did not remove the line underneath it?

Reply:
I actually re-wrote that whole bit of code, I didn't really think about the comment. Should I re-add it?

Reply:
the comment was there for the `mod_stored_nutr()` line, which it does not look like you removed. as such, i don't think removing it was necessary. of course, it may be a superfluous comment, but that's probably nitpicking.

Reply:
I throw it back in anyway, for the sake of having some documentation.